controllers/priorityexpander_controller.go

```diff
@@ -0,0 +1,296 @@
+			"name", r.Configuration.ClusterAutoscalerStatusName,
+		)
+
+		return ctrl.Result{}, err
+	}
+	// Get human readable status from configmap...
+	readableStatus, ok := cm.Data["status"]
+	if !ok {
+		err := fmt.Errorf("invalid autoscaler status configmap")
+		log.Error(
+			err,
+			"unable to parse ClusterAutoscaler status",
+			"namespace", r.Configuration.ClusterAutoscalerStatusNamespace,
+			"name", r.Configuration.ClusterAutoscalerStatusName,
+		)
+		return ctrl.Result{}, err
+	}
+
+	// ... and parse it.
+	status := clusterautoscaler.ParseReadableString(readableStatus)
+
+	var oroot = map[string]map[string]int32{}
+	for _, node := range status.NodeGroups {
+		oroot[node.Name] = make(map[string]int32)
+		oroot[node.Name]["CloudProviderTarget"] = node.Health.CloudProviderTarget
+		oroot[node.Name]["Ready"] = node.Health.Ready
+		oroot[node.Name]["Unready"] = node.Health.Unready
+		oroot[node.Name]["NotStarted"] = node.Health.NotStarted
+		oroot[node.Name]["LongNotStarted"] = node.Health.LongNotStarted
+		oroot[node.Name]["Registered"] = node.Health.Registered
+		oroot[node.Name]["LongUnregistered"] = node.Health.LongUnregistered
+		oroot[node.Name]["MinSize"] = node.Health.MinSize
+		oroot[node.Name]["MaxSize"] = node.Health.MaxSize
+	}
+
+	// Create new PriorityExpander template and parse it
+	t, err := template.New("template").Funcs(sprig.TxtFuncMap()).Parse(pexp.Spec.Template)
+	if err != nil {
+		log.Error(
+			err,
+			"Error parsing PriorityExpander template. Check your syntax and/or rtfm.",
+		)
+		return r.endReconciliation(ctx, log, pexp, controllerutil.OperationResultNone, err)
+	}
+
+	buf := new(bytes.Buffer)
+	if err := t.Execute(buf, oroot); err != nil {
+		log.Error(
+			err,
+			"Error parsing generating template output.",
+		)
+		return r.endReconciliation(ctx, log, pexp, controllerutil.OperationResultNone, err)
+	}
+
+	// parsed content: fmt.Println(buf.String())
+	// Create the new ConfigMap object
+	pecm := kcore_v1.ConfigMap{
+		ObjectMeta: kmeta_v1.ObjectMeta{
+			Name:      r.Configuration.ClusterAutoscalerPEConfigMapName,
+			Namespace: r.Configuration.ClusterAutoscalerStatusNamespace,
+		},
+	}
+
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, &pecm, func() error {
+
+		pecm.Data = map[string]string{
+			"priorities": buf.String(),
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		log.Error(
+			err,
+			"Unable to reconcile ClusterAutoscaler priority expander configmap",
+			"namespace", r.Configuration.ClusterAutoscalerStatusNamespace,
+			"name", r.Configuration.ClusterAutoscalerStatusName,
+		)
+		return r.endReconciliation(ctx, log, pexp, op, err)
+	} else {
+		log.Info("ConfigMap successfully reconciled", "operation", op)
+	}
+
+	return r.endReconciliation(ctx, log, pexp, op, nil)
+}
+
+func (r *PriorityExpanderReconciler) endReconciliation(
+	ctx context.Context,
+	log logr.Logger,
+	pexp corev1alpha1.PriorityExpander,
+	op controllerutil.OperationResult,
+	error error,
+) (ctrl.Result, error) {
```

Comment:
Could be greatly simplified with a status Update instead of Patch (see reconciliation of `Instances`)